### PR TITLE
SPEC-973 : Support filter in listDatabases

### PR DESCRIPTION
**Pull request by rathisekaran**

### CODE REVIEW

File: source/enumerate-databases.rst [hunk 1]
```diff
@@ -237,4 +264,4 @@ all ``sizeOnDisk`` fields in the array of database information documents.
 Changes
 =======
 
-Nothing yet.
+2017-10-30 Adding filter option to listDatabases command
```
Comment (jmikola) [inline source/enumerate-databases.rst]: I'd suggest replacing "Adding" with "Support"

File: source/enumerate-databases.rst [hunk 2]
```diff
@@ -149,6 +174,8 @@ MongoDatabase through MongoClient (e.g. read preference, write concern).
 Drivers SHOULD specify the ``nameOnly`` option when executing the
 ``listDatabases`` command for this method.
 
+Drivers SHOULD implement the ``filter`` option when implementing this method.
```
Comment (jmikola) [inline source/enumerate-databases.rst]: What do you think about "SHOULD support" vs. "SHOULD implement"?

Comment (rathisekaran) [reply source/enumerate-databases.rst]: SHOULD support sounds better.

File: source/enumerate-databases.rst [hunk 3]
```diff
@@ -134,6 +157,8 @@ Older versions of the server that do not support the ``nameOnly`` option for the
 drivers SHOULD always specify the ``nameOnly`` option when they only intend to
 access database names from the ``listDatabases`` command result.
 
+Drivers SHOULD support the ``filter`` option when implementing this method.
```
Comment (behackett) [inline source/enumerate-databases.rst]: I'm not sure I see the use case for the filter option in list_database_names. To see if the names you specified actually exist?

Comment (jmikola) [reply source/enumerate-databases.rst]: Perhaps you just want to list names with a common prefix (maybe relevant for multi-tenant apps).

Comment (behackett) [reply source/enumerate-databases.rst]: Ah, good point.

**Pull request merged**